### DISCUSSION TITLE
Repeatable inside relationship field -  creating PurchaseOrder with Products inside Task creation

### DISCUSSION THREAD

**Discussion by GregaUNK** [Laravel-Backpack/community-forum#1170] [Q&A (Help)]
Hi,

I have a Task CRUD where users can create tasks. During task creation, I would like to add the ability to directly create purchase orders. Each purchase order consists of various text fields and a repeatable field to add multiple products, including pivot fields like quantity, price, etc.

In the Task update view, there should also be an option to view the complete purchase orders and edit them directly.

Do you have any suggestions on how to implement this?

**Comment by jcastroa87** [comment]
Hello @GregaUNK 

I think your best shot is use [relationship Pro with additional data](https://backpackforlaravel.com/docs/6.x/crud-fields#save-additional-data-to-pivot-table)

There you will have the main task fields and inside a relationship field with extra data who will be insert in other Model.

Cheers.

**Reply by GregaUNK** [reply]
Hi @jcastroa87 

Can I have additional pivot relationship fields inside relationship? Like I said I need to add products into purchase order. Products are related to Purchase order with pivot table. This pivot table also includes additional product parameters like quantity, price,... 

Also, how to inline edit related items?

Thanks for help.

**Reply by pxpm** [reply]
Hey @GregaUNK 

If I understood correctly you have a `PurchaseOrderCrud` for the `PurchaseOrderModel`. The `PurchaseOrderModel` has `belongsToMany(ProductModel)` via pivot table ?

In that case, you need to add the additional pivot fields in the relation, `belongsToMany(ProductModel)->withPivot(['quantity', 'price'])`, similar to: https://github.com/Laravel-Backpack/demo/blob/1aad77850c23fc125068442fec225f7f635c25a8/app/Models/Monster.php#L164

And then define the relation field, similar to: https://github.com/Laravel-Backpack/demo/blob/1aad77850c23fc125068442fec225f7f635c25a8/app/Http/Controllers/Admin/MonsterCrudController.php#L1200

If that's not the case, please can you setup a simple repository with only the relevant CRUDs you are trying to create, so that we can try to reproduce the issue ? 

Cheers

**Reply by GregaUNK** [reply]
Hi @pxpm 

I think I haven’t been clear enough.

I need an additional level of relationship. Maybe  this image will help you understand what I want to achieve. I want the user to be able to add and see everything on the TaskCRUD Creation/Update page. I don’t want to redirect them to the PurchaseOrder CRUD to edit the related products or anything else. I want them to stay on the TaskCRUD page and edit everything.

PS: I found this in docs. Maybe this is the problem?

> FIELD USAGE AND RELATIONSHIPS: note that it's not possible to use a repeatable field inside other repeatable field. Relationships that use subfields are under the hood repeatable fields, so the relationship subfields cannot include other repeatable field.

```
  $this->crud->addField([
            'label' => 'Deliveries',
            'name' => 'deliveries',
            'type' => 'relationship',
            'subfields' => [
                ['name' => 'delivery_note_text', 'type' => 'text'],
                [
                    'name'      => 'products',
                    'type' => 'relationship',
                    'label'     => 'Products for delivery',
                    'subfields' => [
                        [
                            'name' => 'quantity',
                            'type' => 'number',
                            'label' => 'Quantity',
                            'attributes' => ["step" => "0.01"],
                            'wrapper' => ['class' => 'form-group col-md-3'],
                            'default' => 1
                        ]
                    ],
                    'pivotSelect' => [
                        'label' => 'Products',
                        'placeholder' => 'Pick a product',
                        'wrapper' => ['class' => 'form-group col-md-6'],

                        'ajax' => true,
                        'attribute' => 'full_name', // attribute on Product model that is shown to user
                        'data_source' => '/api/product/getSearchProducts?whids=4',
                    ],
                    //      'disabled' => 'disabled',
                    'new_item_label' => 'Add product', // Label for the button to add a new product entry
                    'init_rows' => 0, // Number of rows when first loading the page
                ]
            ]
        ]);
```

![Screenshot 2024-10-06 at 08 46 50](https://github.com/user-attachments/assets/f96cc945-0165-4e12-92f9-8078032f4d64)

**Comment by pxpm** [comment]
Ohh totally @GregaUNK repeatable over repeatable is not possible atm. 

If you explore that possibility, let us know what errors you are facing and what are you doing to "overcome" them. 

We haven't explored that since we created the field, but we changed and fixed so many things, that the difficulties we had before, may not all apply now, but we didn't went back to try to make it work. 

Maybe try using the table field inside the repeatable ? 

Cheers

**Comment by jcastroa87** [comment]
Due to not activity, I will close this issue, but please feel free to re-open or create a new one if needed.

Cheers.
